### PR TITLE
Add Swagger UI docs

### PR DESCRIPTION
**Pull request by shakedlv**
Implemented Swagger UI  including auth0 authentication for API

### CODE REVIEW

**Review by galdd — APPROVED** [review]
Thanks for the work on this. Everything looks good to me. Approved.

File: src/index.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ app.use(cors());
 app.use(express.json()); // Used to parse JSON bodies
 app.use(express.urlencoded()); // Parse URL-encoded bodies using query-string library
 
+
```
Comment (ofir-zeitoun) [inline src/index.ts:15]: רקצםהק

File: src/routes/index.ts [hunk 2]
```diff
@@ -10,11 +11,17 @@ import {
   checkJwt,
 } from "./middlewares";
 import users from "../features/users";
+import { swaggerDocument, swaggerOptions } from "../swagger";
 
 export const routes = (app: Express) => {
   app.get(...healthCheck);
   app.use(...stats);
-  app.use(checkJwt, () => {});
+  app.use(
```
Comment (ofir-zeitoun) [inline src/routes/index.ts:19]: make as the rest:
`...swagger`

File: src/swagger.ts [hunk 3]
```diff
@@ -0,0 +1,757 @@
...
+        "description": "Swagger for Expenses API",
+        "version": "1.0.0"
+    },
+    "externalDocs": {
+        "description": "Server Github",
+        "url": "https://github.com/ofir-zeitoun/expenses-server"
+    },
+    "servers": [
+        {
+            "url": "http://localhost:1337"
```
Comment (ofir-zeitoun) [inline src/swagger.ts:24]: should it be localhost?

File: src/swagger.ts
Comment (ofir-zeitoun) [inline src/swagger.ts:1]: split to domain files - make `domain.swagger.ts` and import from it

File: src/swagger.ts (hunk 3 above)
Comment (shakedlv) [reply src/swagger.ts:24]: I could not find any answer about which one is the best practice
All tutorials and documentation found use localhost or an API server URL (example.com), and not IP

File: src/features/expenses/expenses.swagger.ts [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            "summary": "Get All Expenses",
+            "description": "Retrieves a list of all expenses.",
+            "tags": [
+                "expenses"
+            ],
+            "responses": {
+                "200": {
+                    "description": "OK",
+                    "content": {
+                        "application/json": {}
```
Comment (ofir-zeitoun) [inline src/features/expenses/expenses.swagger.ts:13]: should be empty?

File: src/features/expenses/expenses.swagger.ts [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+                    "description": "Created",
+                    "content": {
+                        "application/json": {
+                            "schema": {
+                                "$ref": "#/components/schemas/Expense"
+                            }
+                        }
+                    }
+                },
+                "400": {
```
Comment (ofir-zeitoun) [inline src/features/expenses/expenses.swagger.ts:45]: can reuse?

File: src/features/expenses/expenses.swagger.ts
Comment (ofir-zeitoun) [inline src/features/expenses/expenses.swagger.ts:1]: split to:
* expenses.swagger.api.ts
* expenses.swagger.schema

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**